app/forms/Location.scala

```diff
@@ -16,37 +16,56 @@
 
 package forms
 
-import forms.Mapping.requiredRadio
 import play.api.data.Forms.text
 import play.api.data.{Form, Forms}
-import play.api.libs.json.Json
+import play.api.libs.json.{Json, OFormat}
 import services.Countries.allCountries
 import utils.validators.forms.FieldValidator._
 
-case class Location(locationType: String, qualifierCode: String, locationCode: String, country: String) {
-
-  def asString: String = country + locationType + qualifierCode + locationCode
-}
+case class Location(code: String)
 
 object Location {
-  implicit val format = Json.format[Location]
+
+  implicit val format: OFormat[Location] = Json.format[Location]
 
   val formId = "Location"
 
   val correctLocationType: Set[String] = Set("A", "B", "C", "D")
-
   val correctQualifierCode: Set[String] = Set("U", "Y")
 
+  /**
+    * Country is in two first characters in Location Code
+    */
+  private val validateCountry: String => Boolean = (input: String) => {
+    val countryCode = input.take(2)
+    allCountries.exists(_.countryCode == countryCode)
+  }
+
+  /**
+    * Location Type is defined as third character in Location Code
+    */
+  private val validateLocationType: String => Boolean = (input: String) => {
```

Comment:
What do you think about moving `correctLocationType` here inside definition

Reply:
Solved